OpenRA.Game/Traits/Target.cs

```diff
@@ -165,6 +165,35 @@ public IEnumerable<WPos> Positions
 			}
 		}
 
+		public IEnumerable<WPos> AttackablePositions
+		{
+			get
+			{
+				switch (Type)
+				{
+					case TargetType.Actor:
+						if (!actor.Targetables.Any(Exts.IsTraitEnabled))
+							return new[] { actor.CenterPosition };
+
+						var attackablePositions = actor.TraitsImplementing<IAttackablePositions>();
+						if (attackablePositions.Any())
+						{
+							// TODO: Add the AttackablePositions of all traits implementing IAttackablePositions
+							return attackablePositions.First().AttackablePositions(actor);
```

Comment:
This compiles, and seems to work in RA, at least.
```csharp
if (attackablePositions.Any())
{
	// TODO: Add the AttackablePositions of all traits implementing IAttackablePositions
	var target = this;
	return attackablePositions.SelectMany(ap => ap.AttackablePositions(target.actor));
}
```

That workaround is required because `Target` is a struct, not a class.